test/integration/datasette_metadata_test.py

```diff
@@ -0,0 +1,85 @@
+"""Test the metadata.yml file outputted for Datasette."""
+
+import json
+import logging
+import shutil
+from pathlib import Path
+
+import datasette.utils
+import pytest
+import yaml
+
+# import subprocess
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.script_launch_mode('inprocess')
+def test_metadata_script(script_runner, pudl_settings_fixture, pudl_engine):
+    """Run metadata_to_yml for testing."""
+    metadata_yml = Path(pudl_settings_fixture["pudl_out"], "metadata.yml")
+    logger.info(f"Writing Datasette Metadata to {metadata_yml}")
+
+    ret = script_runner.run(
+        'metadata_to_yml',
+        '-o',
+        str(metadata_yml),
+        print_result=False,
+    )
+    assert ret.success
+
+    logger.info("Parsing generated metadata using datasette utils.")
+    metadata_json = json.dumps(yaml.safe_load(metadata_yml.open()))
+    # The parsed_metadata dictionary would be the thing to spot check, to see
+    # if values / elements are as expected:
+    parsed_metadata = datasette.utils.parse_metadata(metadata_json)
+    # Some examples:
+    assert set(parsed_metadata['databases']) == set(["pudl", "ferc1"])
+    assert parsed_metadata["license"] == "CC-BY-4.0"
+    assert parsed_metadata['databases'][
+        'pudl']['source_url'] == 'https://github.com/catalyst-cooperative/pudl'
+    assert parsed_metadata['databases'][
+        'pudl']['tables']['plants_entity_eia']['label_column'] == 'plant_name_eia'
+    for tbl_name in parsed_metadata['databases']['pudl']['tables']:
+        assert parsed_metadata['databases'][
+            'pudl']['tables'][tbl_name]['columns'] is not None
```

Comment:
If there's an **easy** programmatic way to check that the expected tables and columns are showing up in the output metadata, that could be good. It looks like there's quite a bit of text munging (escaping characters, etc.) that happens in the conversion to JSON, so probably the descriptions or longer form elements aren't going to be easy to compare directly. But if you re-create the same `DatasetteMetadata` object in here that's being used to generate the output metadata, and then loop through and verify that for each table, all the expected columns are present that would probably catch a lot of errors. It would need to be easy to keep the `DatasetteMetadata` object used here, and by the script itself in sync though.